### PR TITLE
fix: fix inline scripts and styles potentially breaking

### LINKED ISSUE #1: Inline scripts can get broken
### Details

Hi @DylanPiercey, awesome small library you've created 😄 

Anyways one thing I noticed as I was playing with it is that inline scripts don't seem to be fully supported, what I mean is that if an inline script gets truncated in the stream, then it won't be correctly handled by the library, as the library will either execute only part of the script and/or will present a syntax error depending on where the script got truncated.

For example I have created the following Cloudflare worker which exposes a stream which truncates a very simple inline script:

```js
addEventListener("fetch", event => {
  event.respondWith(handleRequest(event.request))
})

async function handleRequest(request) {
  return new Response(getStream()});
}

function getStream() {
  const { writable, readable } = new TransformStream();
  writeInStream(writable);
  return readable;
}

const chunks = [
`<script>
  console.log('`,`this is streaming!');
</script>`,
];

async function writeInStream(writable) {
  const encoder = new TextEncoder();
  const writer = writable.getWriter();
  const write = str => writer.write(encoder.encode(str));

  for (const chunk of chunks) {
    write(chunk);
    await new Promise(resolve => setTimeout(resolve, 1000));
  }

  writer.close();
}
```

(well, the implementation there is not important, the relevant thing is that the stream is comprised of two separate chunks:
`<script>console.log('` and `this is streaming!');</script>`)

If I run the code you present in the Repo's readme (in which of course I replace `"https://ebay.com"` with the worker's URL)
I get a syntax error in the console:
Screenshot 2022-09-14 at 22 44 51

The issue being that initially the `script` gets only partially added to the document:
Kapture 2022-09-14 at 22 48 38

I think that the script's content should be buffered and that the script should be added fully to the document in a single operation, so that the whole script will be evaluated correctly by the browser.

Naturally my example here is a silly one, but I came across this issue as I was trying to use the library and the stream I was trying to add to the dom happens to get truncated causing a syntax error and making the whole thing not work.

Have you considered this issue? and/or do you have an opinion/potential solution for it?

As I'm interested in what this library is doing, I wouldn't mind trying to add a fix and submit a PR 🙂

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
fix the issue which breaks scripts split in different chunks in the input stream
by applying them only after the scripts has been fully visited

also fix the same issue for embeded styles, even though there everything works
fine it is better to apply all the styles together and avoid potential flashes of unexpected style combinations

resolves #1

## Screenshots (if appropriate):

This is the result:
Kapture 2022-09-15 at 15 45 29

## Checklist:

- [ ] I have updated/added documentation affected by my changes.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -178,3 +189,24 @@ function getPreloadLink(node: any) {
...
+  pendingText: Text | null,
+  inlineTextHostNode: Node | null
+) {
+  if (pendingText && inlineTextHostNode) {
+    inlineTextHostNode.appendChild(pendingText);
+  }
+}
+
+function isInlineHost(node: Node | null) {
+  if (node?.nodeType !== Node.ELEMENT_NODE) {
```
Comment (DylanPiercey) [inline src/index.ts]: Could be mistaken, but isn't `node` guaranteed to be defined here? We could skip the null check if so

**Comment by DylanPiercey** [comment]
Thanks for taking the time to put up the PR! I think it looks pretty good 👍

File: src/index.ts (hunk 1 above)
Comment (dario-piotrowicz) [reply src/index.ts]: ah yeah, sorry my bad 🙇 , I've added a non-null assertion instead in the callers instead

File: src/index.ts [hunk 2]
```diff
@@ -23,12 +23,13 @@ export = function writableDOM(
...
   let resolve: void | (() => void);
   let isBlocked = false;
+  let inlineHostNode: Node | null = null;
 
   return {
     write(chunk: string) {
       doc.write(chunk);
 
-      if (pendingText) {
+      if (pendingText && isInlineHost(pendingText.parentElement!)) {
```
Comment (IgorMinar) [inline src/index.ts]: could you just check if `inlineHostNode !== null`?

Comment (dario-piotrowicz) [reply src/index.ts]: yes, I've applied the change, sorry, silly mistake 😅, thank you very much! 😄

File: src/index.ts [hunk 3]
```diff
@@ -178,3 +189,24 @@ function getPreloadLink(node: any) {
...
+  inlineTextHostNode: Node | null
+) {
+  if (pendingText && inlineTextHostNode) {
+    inlineTextHostNode.appendChild(pendingText);
+  }
+}
+
+function isInlineHost(node: Node) {
+  if (node.nodeType !== Node.ELEMENT_NODE) {
+    return false;
```
Comment (dario-piotrowicz) [inline src/index.ts]: this line is not being covered by tests, and by checking the [MDN nodeType docs](https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType) I think that the node here can't be anything else but an Element, so I am just going to remove this check, please let me know if you disagree

**Comment by dario-piotrowicz** [comment]
@DylanPiercey I've added a new very simple test because it looked like my change was causing some tests coverage failure (even though the non-covered lines seem unrelated to my change 😕), I hope it's fine, if you could approve the workflow we can see if now all is green 😁🤞

**Comment by DylanPiercey** [comment]
Appreciate the changes and additional tests. Looks good to me!

**Comment by dario-piotrowicz** [comment]
> Appreciate the changes and additional tests. Looks good to me!

Awesome! thanks! 😄

**Pull request merged**